### ISSUE TITLE
Allow CI runs to disable listing dependencies

### ISSUE THREAD

**Issue by lengau** [tox-dev/tox#3023]
## What's the problem this feature will solve?

Right now CI runs always list dependencies for each environment. While this is often nice, it can result in excessive output, especially when using tox to run linters and other tools.

## Describe the solution you'd like

If `list_dependencies_command` is explicitly set as an empty string for an environment, skip the listing dependencies command.

## Alternative Solutions

One alternative would be a separate configuration option for an environment to specifically disable listing dependencies.

## Additional context

snapcraft and several other projects use tox to run our linters. However, when running in CI, it can be difficult to find specific linting failures between all the `pip freeze` lines:

![image](https://github.com/tox-dev/tox/assets/4305943/1a51f38a-74ec-4fac-925d-736db96b4301)

**Comment by masenf** [comment]
> If list_dependencies_command is explicitly set as an empty string for an environment, skip the listing dependencies command.

This sounds like a good solution to me.

**Comment by gaborbernat** [comment]
I'd would rather have a flag for it than ask the user to change the configuration.

**Comment by masenf** [comment]
Fair enough. An explicit --list-dependencies / --no-list-dependencies on the cli is probably useful. I've actually exported CI=1 in non-CI runs just to get the pip freeze output.

**Comment by lengau** [comment]
Allowing `list_dependencies_command` to be empty ended up being more invasive than I expected. I've got a PR using the `--list-dependencies` CLI options coming as it was pretty easy to implement.
